modules/smtp/modules.php

```diff
@@ -407,8 +408,24 @@ public function process() {
  * @subpackage smtp/handler
  */
 class Hm_Handler_add_smtp_servers_to_page_data extends Hm_Handler_Module {
+
+    private function generateId($servers) {
+        $changed = false;
+        foreach ($servers as $index => $server) {
+            if (!array_key_exists('id', $server)) {
+                $servers[$index]['id'] = uniqid();
+                $changed = true;
+            }
+        }
+        if ($changed) {
+            $this->session->record_unsaved('SMTP server ID updated');
+        }
+        return $servers;
+    }
+
     public function process() {
         $servers = Hm_SMTP_List::dump();
+        $servers = $this->generateId($servers);
```

Comment:
This is good that we generate the IDs if we have cached server data from config. What about the rest of the lists - pop3, imap, profile, feed?